testsuite/Main.hs

```diff
@@ -40,6 +47,79 @@ import           Test.Tasty.HUnit
 #define NEEDS_PRIMS_GHC(x) (x)
 #endif
 
+-- | GHC version as major.minor.patch1. For example: 8.10.2.
+ghcVersion3 :: String
+ghcVersion3 =
+  let ghc_p1 = __GLASGOW_HASKELL_PATCHLEVEL1__ in
+  intercalate "." (map show (versionBranch compilerVersion <> [ghc_p1]))
+
+-- | Set GHC_PACKAGE_PATH for local Cabal install. Currently hardcoded for Unix.
+setCabalPackagePaths :: IO ()
+setCabalPackagePaths = do
+  cabalHome <- fmap homeDirectory (getUserEntryForName =<< getLoginName)
+  here <- getCurrentDirectory
+  setEnv "GHC_PACKAGE_PATH" $
+       cabalHome <> "/.cabal/store/ghc-" <> ghcVersion3 <> "/package.db"
+    <> ":"
+    <> here <> "/dist-newstyle/packagedb/ghc-" <> ghcVersion3
+    <> ":"
+
+-- | See 'compiledWith'
+data RunWith
+  = Stack
+  | Cabal
+  | Global
+  deriving (Show)
+
+-- | Detects Clash binary the testsuite should use (in order):
+--
+--     * If STACK_EXE is present, use Stack's Clash
```

Comment:
Is it a good idea to look just for `STACK_EXE`, or should we care if there is a `.stack-work` directory?

Reply:
Hmm, rephrase: I find it strange on first read that we care about an envvar for stack, and the presence of a directory for cabal? What if I built with cabal but also have stack on my system - will it always choose stack?

Reply:
The logic is: 

* `STACK_EXE` should only be set when running with `stack run testsuite`. 
* If you intend to run the testsuite with a globally installed clash, the testsuite binary is (probably?) also installed globally. I didn't find a way to detect whether we're running in `cabal run`, hence the logic of looking at the presence or absence of `dist-newstyle`.

Reply:
Right, that makes sense. So the goal is to make sure you use the same `clash` when spawning processes as the `clash` as the tool running the testsuite would use?

Reply:
Yes.